### PR TITLE
[735]reset campaign name when changing from ETP

### PR DESCRIPTION
**Pull request by goodviber**
### Context
We need to make sure we're resetting the campaign_name back to nil when a user changes from ETP to a different subject.
What needs to be done

Handle this edge case where ever the subject is changed so that the field is set back to nil on the course
### Changes proposed in this pull request
clear campaign_name params for new action and update course model in update action in subjects controller
### Guidance to review
Create new Physics course then change course from confirmation page, then change back to Physics course, ETP radio buttons should be blank. Do same with existing course from basics details page.
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by goodviber** [comment]
[Reset campaign name field when changing from ETP](https://trello.com/c/7UCW31IK/735-reset-campaign-name-field-when-changing-from-etp)

File: spec/features/publish/courses/new_engineers_teach_physics_spec.rb [hunk 1]
```diff
@@ -29,6 +29,39 @@
...
+    and_i_click_continue
+    then_i_am_met_with_the_age_range_page_with_latin(:secondary, :latin)
+
+    when_i_go_back
+    and_i_select_a_subject(:physics)
+    and_i_click_continue
+    then_i_am_met_with_the_engineers_teach_physics_page_with_no_etp(:secondary, :physics)
+    and_i_click_continue
+    then_i_see_an_error_message
+  end
```
Comment (avinhurry) [inline spec/features/publish/courses/new_engineers_teach_physics_spec.rb:63]: Do we need this test if we have the other one? Is it testing anything that the other one isn't? It's a big feature test and our test suite is already quite slow!

**Review by avinhurry — APPROVED** [review]
Looks good to me, left a couple of comments at your discretion 👍🏼

File: spec/features/publish/courses/new_engineers_teach_physics_spec.rb (hunk 1 above)
Comment (goodviber) [reply spec/features/publish/courses/new_engineers_teach_physics_spec.rb:63]: The new path validates in the controller, the edit path validates in the form_object, so I think we should cover both

**Pull request merged**